tests/integration_test.rs

```diff
@@ -362,3 +398,85 @@ async fn test_custom_env_name(options: Options) {
     assert!(unpack_result.is_ok(), "{:?}", unpack_result);
     assert!(env_dir.is_dir());
 }
+
+#[rstest]
+#[tokio::test]
+async fn test_run_packed_executable(options: Options, required_fs_objects: Vec<&'static str>) {
+    let temp_dir = tempfile::tempdir().unwrap();
+    let mut pack_options = options.pack_options;
+    pack_options.create_executable = true;
+
+    #[cfg(target_os = "windows")]
+    {
+        pack_options.output_file = temp_dir.path().join("environment.ps1");
+    }
+    #[cfg(any(target_os = "linux", target_os = "macos"))]
+    {
+        pack_options.output_file = temp_dir.path().join("environment.sh");
+    }
+
+    let pack_file = pack_options.output_file.clone();
+
+    let pack_result = pixi_pack::pack(pack_options).await;
+    assert!(pack_result.is_ok(), "{:?}", pack_result);
+
+    assert!(
+        pack_file.exists(),
+        "Pack file does not exist at {:?}",
+        pack_file
+    );
+
+    #[cfg(any(target_os = "linux", target_os = "macos"))]
+    {
+        assert_eq!(pack_file.extension().unwrap(), "sh");
+        let output = Command::new("bash")
+            .arg(&pack_file)
+            .arg("-o")
+            .arg(options.output_dir.path())
+            .output()
+            .expect("Failed to execute packed file for extraction");
+        assert!(
+            output.status.success(),
+            "Packed file execution failed: {:?}",
+            output
+        );
+    }
+    #[cfg(target_os = "windows")]
+    {
+        assert_eq!(pack_file.extension().unwrap(), "ps1");
+        let output = Command::new("powershell")
+            .arg("-File")
+            .arg(&pack_file)
+            .arg("-o")
+            .arg(options.output_dir.path())
+            .output()
+            .expect("Failed to execute packed file for extraction");
+        assert!(output.status.success(), "Packed file execution failed");
+    }
```

Comment:
let's switch the order here, in the other examples you always had windows before unix